plugin crashes while refreshing state of an app previously created

I am using the new changes from master branch and was happy to find the previous issue of service binding and service creation wait time and even issue with app being pushed but not started were fixed, however once an app is pushed successfully and on a second run where terraform refreshes the app state it crashes, the plugin successfully gets a refresh state on existing spaces, services, routes, ups, but fails on a refresh on an existing app that was pushed, below trace log happens on an app that was successfully created on first execution but crashes on a refresh on a second execution

this is the code
```
resource "cloudfoundry_route" "config_service" {
	domain = "${data.cloudfoundry_domain.us-east.id}"
	space = "${cloudfoundry_space.s1.id}"
	hostname = "${var.cf_space}-config-service"
}

resource "cloudfoundry_app" "config_service" {
    name        = "config_service_web"
    space       = "${cloudfoundry_space.s1.id}"
    memory      =   "512"
    disk_quota  =   "1024"
    route       = { 
       default_route = "${cloudfoundry_route.config_service.id}"
    }
    docker_image = "${var.docker_image_url}"
    docker_credentials = {
        username = "${var.docker_reg_username}"
        password = "${var.docker_reg_password}"
    }
    service_binding = {
           service_instance = "${cloudfoundry_user_provided_service.logdrainer.id}" 
    }
    timeout         = 1800
    environment = {
        SPRING_APPLICATION_JSON = "${var.config_service_vault_params}"
        MANAGEMENT_ENDPOINTS_WEB_EXPOSURE_INCLUDE = "health,info,metrics,prometheus,loggers,env"
    }
    enable_ssh = true
    stopped    = false
}
```

below is part of the trace log

```$ TF_LOG=trace terraform plan -var-file=dev.tfvars```

```
2018/10/11 15:09:38 [TRACE] [walkRefresh] Exiting eval tree: cloudfoundry_route.config_service
2018/10/11 15:09:38 [TRACE] dag/walk: walking "cloudfoundry_app.config_service"
2018/10/11 15:09:38 [TRACE] vertex 'root.cloudfoundry_app.config_service': walking
2018/10/11 15:09:38 [TRACE] vertex 'root.cloudfoundry_app.config_service': evaluating
2018/10/11 15:09:38 [TRACE] [walkRefresh] Entering eval tree: cloudfoundry_app.config_service
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalSequence
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalInterpolate
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalCountCheckComputed
cloudfoundry_app.config_service: Refreshing state... (ID: 574f316b-687d-48f8-a9b2-ca7789352c8d)
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalIf
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalCountFixZeroOneBoundary
2018/10/11 15:09:38 [TRACE] [walkRefresh] Exiting eval tree: cloudfoundry_app.config_service
2018/10/11 15:09:38 [TRACE] vertex 'root.cloudfoundry_app.config_service': expanding/walking dynamic subgraph
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.ResourceCountTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [TRACE] OrphanResourceCount: Starting...
2018/10/11 15:09:38 [TRACE] OrphanResourceCount: Checking: cloudfoundry_app.config_service
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.OrphanResourceCountTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [DEBUG] Attaching resource state to "cloudfoundry_app.config_service": &terraform.ResourceState{Type:"cloudfoundry_app", Dependencies:[]string{"cloudfoundry_route.config_service", "cloudfoundry_space.s1", "cloudfoundry_user_provided_service.phe-logdrainer"}, Primary:(*terraform.InstanceState)(0xc420328370), Deposed:[]*terraform.InstanceState{}, Provider:"provider.cloudfoundry", mu:sync.Mutex{state:0, sema:0x0}}
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.AttachStateTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.TargetsTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [DEBUG] ReferenceTransformer: "cloudfoundry_app.config_service" references: []
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.ReferenceTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [TRACE] Graph after step *terraform.RootTransformer:

cloudfoundry_app.config_service - *terraform.NodeRefreshableManagedResourceInstance
2018/10/11 15:09:38 [TRACE] dag/walk: added new vertex: "cloudfoundry_app.config_service"
2018/10/11 15:09:38 [TRACE] dag/walk: walking "cloudfoundry_app.config_service"
2018/10/11 15:09:38 [TRACE] vertex 'root.cloudfoundry_app.config_service': walking
2018/10/11 15:09:38 [TRACE] vertex 'root.cloudfoundry_app.config_service': evaluating
2018/10/11 15:09:38 [TRACE] [walkRefresh] Entering eval tree: cloudfoundry_app.config_service
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalSequence
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalGetProvider
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalReadState
2018/10/11 15:09:38 [TRACE] root: eval: *terraform.EvalRefresh
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry:
Error: Error refreshing state: 1 error(s) occurred:

* cloudfoundry_app.config_service: 1 error(s) occurred:

* cloudfoundry_app.config_service: cloudfoundry_app.config_service: unexpected EOF


panic: interface conversion: interface {} is nil, not float64
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry:
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: goroutine 707 [running]:
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).readRouteMappings.func1(0x101df20, 0xc00014eed0, 0xc0004e0000)
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:257 +0x8ec
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/code.cloudfoundry.org/cli/cf/net.Gateway.ListPaginatedResources(0x138ad20, 0xc0000e6790, 0x12239f8, 0x0, 0x12a05f200, 0x0, 0x0, 0x0, 0x7f4532e26050, 0xc0002b0900, ...)
2018-10-11T15:09:39.389Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/code.cloudfoundry.org/cli/cf/net/gateway.go:155 +0x315
2018-10-11T15:09:39.390Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).readRouteMappings(0xc000788410, 0xc0006beb10, 0x24, 0x11d3de0, 0x8, 0x0, 0x0, 0x0, 0x1e3f2c8, 0xc000406160)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:243 +0x258
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).ReadRouteMappingsByApp(0xc000788410, 0xc0006beb10, 0x24, 0x16, 0xc0006beb70, 0x24, 0xc0005e68e0, 0xc00052b988)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:235 +0x58
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry.resourceAppRead(0xc00072a150, 0x117ed20, 0xc0001b6c80, 0xc00072a150, 0x0)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/resource_cf_app.go:568 +0x2c3
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc0002912d0, 0xc000184140, 0x117ed20, 0xc0001b6c80, 0xc000143218, 0x4c2701, 0xf8d4c0)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:354 +0x160
2018-10-11T15:09:39.392Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc000291340, 0xc0001840a0, 0xc000184140, 0xc000057500, 0x53380d134d2c6dfc, 0x7f4532ee8d80)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc0002a1060, 0xc000406070, 0xc000406140, 0x0, 0x0)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry: reflect.Value.call(0xc000385620, 0xc0001735b8, 0x13, 0x11cf283, 0x4, 0xc000538f18, 0x3, 0x3, 0xc000156d00, 0xc0002ddd08, ...)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/reflect/value.go:447 +0x449
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: reflect.Value.Call(0xc000385620, 0xc0001735b8, 0x13, 0xc000486f18, 0x3, 0x3, 0x1, 0x6607fe, 0xc000035f01)
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/reflect/value.go:308 +0xa4
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: net/rpc.(*service).call(0xc0002e5840, 0xc00001b400, 0xc0000b7b30, 0xc0000b7b40, 0xc00048e580, 0xc0006409e0, 0xf8d480, 0xc000406070, 0x16, 0xf8d4c0, ...)
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/net/rpc/server.go:384 +0x14e
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: created by net/rpc.(*Server).ServeCodec
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/net/rpc/server.go:481 +0x47e
2018/10/11 15:09:39 [ERROR] root: eval: *terraform.EvalRefresh, err: cloudfoundry_app.config_service: unexpected EOF
2018/10/11 15:09:39 [ERROR] root: eval: *terraform.EvalSequence, err: cloudfoundry_app.config_service: unexpected EOF
2018/10/11 15:09:39 [TRACE] [walkRefresh] Exiting eval tree: cloudfoundry_app.config_service
2018/10/11 15:09:39 [TRACE] dag/walk: upstream errored, not walking "provider.cloudfoundry (close)"
2018/10/11 15:09:39 [TRACE] dag/walk: upstream errored, not walking "root"
2018/10/11 15:09:39 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-10-11T15:09:39.418Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-10-11T15:09:39.427Z [DEBUG] plugin.terraform-provider-hsdp: 2018/10/11 15:09:39 [ERR] plugin: plugin server: accept unix /tmp/plugin903295768: use of closed network connection
2018-10-11T15:09:39.434Z [DEBUG] plugin: plugin process exited: path=/tffiles/terraform.d/plugins/linux_amd64/terraform-provider-cloudfoundry
2018-10-11T15:09:39.439Z [DEBUG] plugin: plugin process exited: path=/tffiles/terraform.d/plugins/linux_amd64/terraform-provider-hsdp
panic: interface conversion: interface {} is nil, not float64
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry:
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: goroutine 707 [running]:
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).readRouteMappings.func1(0x101df20, 0xc00014eed0, 0xc0004e0000)
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:257 +0x8ec
2018-10-11T15:09:39.388Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/code.cloudfoundry.org/cli/cf/net.Gateway.ListPaginatedResources(0x138ad20, 0xc0000e6790, 0x12239f8, 0x0, 0x12a05f200, 0x0, 0x0, 0x0, 0x7f4532e26050, 0xc0002b0900, ...)
2018-10-11T15:09:39.389Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/code.cloudfoundry.org/cli/cf/net/gateway.go:155 +0x315
2018-10-11T15:09:39.390Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).readRouteMappings(0xc000788410, 0xc0006beb10, 0x24, 0x11d3de0, 0x8, 0x0, 0x0, 0x0, 0x1e3f2c8, 0xc000406160)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:243 +0x258
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi.(*RouteManager).ReadRouteMappingsByApp(0xc000788410, 0xc0006beb10, 0x24, 0x16, 0xc0006beb70, 0x24, 0xc0005e68e0, 0xc00052b988)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi/route_manager.go:235 +0x58
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/cloudfoundry.resourceAppRead(0xc00072a150, 0x117ed20, 0xc0001b6c80, 0xc00072a150, 0x0)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/cloudfoundry/resource_cf_app.go:568 +0x2c3
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc0002912d0, 0xc000184140, 0x117ed20, 0xc0001b6c80, 0xc000143218, 0x4c2701, 0xf8d4c0)
2018-10-11T15:09:39.391Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:354 +0x160
2018-10-11T15:09:39.392Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc000291340, 0xc0001840a0, 0xc000184140, 0xc000057500, 0x53380d134d2c6dfc, 0x7f4532ee8d80)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry: github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc0002a1060, 0xc000406070, 0xc000406140, 0x0, 0x0)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /go/src/github.com/terraform-providers/terraform-provider-cf/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry: reflect.Value.call(0xc000385620, 0xc0001735b8, 0x13, 0x11cf283, 0x4, 0xc000538f18, 0x3, 0x3, 0xc000156d00, 0xc0002ddd08, ...)
2018-10-11T15:09:39.393Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/reflect/value.go:447 +0x449
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: reflect.Value.Call(0xc000385620, 0xc0001735b8, 0x13, 0xc000486f18, 0x3, 0x3, 0x1, 0x6607fe, 0xc000035f01)
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/reflect/value.go:308 +0xa4
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: net/rpc.(*service).call(0xc0002e5840, 0xc00001b400, 0xc0000b7b30, 0xc0000b7b40, 0xc00048e580, 0xc0006409e0, 0xf8d480, 0xc000406070, 0x16, 0xf8d4c0, ...)
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/net/rpc/server.go:384 +0x14e
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry: created by net/rpc.(*Server).ServeCodec
2018-10-11T15:09:39.395Z [DEBUG] plugin.terraform-provider-cloudfoundry:        /usr/local/go/src/net/rpc/server.go:481 +0x47e
2018/10/11 15:09:39 [ERROR] root: eval: *terraform.EvalRefresh, err: cloudfoundry_app.config_service: unexpected EOF
2018/10/11 15:09:39 [ERROR] root: eval: *terraform.EvalSequence, err: cloudfoundry_app.config_service: unexpected EOF
2018/10/11 15:09:39 [TRACE] [walkRefresh] Exiting eval tree: cloudfoundry_app.config_service
2018/10/11 15:09:39 [TRACE] dag/walk: upstream errored, not walking "provider.cloudfoundry (close)"
2018/10/11 15:09:39 [TRACE] dag/walk: upstream errored, not walking "root"
2018/10/11 15:09:39 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-10-11T15:09:39.418Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-10-11T15:09:39.427Z [DEBUG] plugin.terraform-provider-hsdp: 2018/10/11 15:09:39 [ERR] plugin: plugin server: accept unix /tmp/plugin903295768: use of closed network connection
2018-10-11T15:09:39.434Z [DEBUG] plugin: plugin process exited: path=/tffiles/terraform.d/plugins/linux_amd64/terraform-provider-cloudfoundry
2018-10-11T15:09:39.439Z [DEBUG] plugin: plugin process exited: path=/tffiles/terraform.d/plugins/linux_amd64/terraform-provider-hsdp



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/cloudfoundry/cfapi/route_manager.go
+++ b/cloudfoundry/cfapi/route_manager.go
@@ -254,7 +254,9 @@ func (rm *RouteManager) readRouteMappings(id, key string) (mappings []map[string
 			mapping["route"] = routeResource["entity"].(map[string]interface{})["route_guid"].(string)
 		}
 		if v, ok := routeResource["entity"].(map[string]interface{})["app_port"]; ok {
-			mapping["port"] = int(v.(float64))
+			if port, ok := v.(float64); ok {
+				mapping["port"] = int(port)
+			}
 		}
 		mappings = append(mappings, mapping)
 		return true

